load_tests/load_tests/runner.py

```diff
@@ -0,0 +1,55 @@
+import time
+from typing import Any, List
+
+from locust.events import request_success
+from rasa_integration_testing.interaction import Interaction, InteractionLoader
+from rasa_integration_testing.runner import ScenarioRunner
+from rasa_integration_testing.scenario import Scenario, ScenarioFragmentLoader
+from socketio import Client
+
+from load_tests.constants import SESSION_ID_KEY
+
+EVENT_BOT_UTTERED = "bot_uttered"
+EVENT_USER_UTTERED = "user_uttered"
+
+
+class LocustRunner(ScenarioRunner):
+    def __init__(
+        self,
+        interaction_loader: InteractionLoader,
+        scenario_fragment_loader: ScenarioFragmentLoader,
+    ):
+        self._interaction_loader = interaction_loader
+        self._scenario_fragment_loader = scenario_fragment_loader
+
+    def run(self, scenario: Scenario, client: Client, session_id: str) -> None:
+        interactions: List[Interaction] = self._resolve_interactions(scenario)
+
+        start_time = time.time()
+
+        @client.event
+        def bot_uttered(data):
+            response_time = int((time.time() - start_time) * 1000) #epochs to ms
+            request_success.fire(
+                request_type="socketio received",
+                name=EVENT_BOT_UTTERED,
+                response_time=response_time,
+                response_length=len(str(data)),
+            )
+            if len(interactions) > 0:
+                user_uttered(interactions.pop(0))
```

Comment:
I'm not really familiar with our `rasa_integration_testing` framework. is there any validation here that the bot answered the with the correct response? I guess the test should fail if it does not go as described by the scenario. That might even indicate incorrect behavior on high load.

Reply:
Now that I think about it - we know that, when encountering an issue within an action (which may be the case on high load) - Rasa will just continue the dialogue answering with something else than what's expected. I think this is the kind of error we want to be aware of while running load test.

Reply:
The testing framework doesn't properly support socketio as of now. We will run the integration tests on the side for now to validate that heavy load doesn't impact routes but this is not an optimal solution since it doesn't use socketio and the load tests as a base.